client/components/relatedProducts/RelatedProducts.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+
+import Card from './Card';
+
+import '../../css/relatedProducts/RelatedProducts.scss';
+
+const RelatedProducts = ({ productId }) => {
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    // make a GET request to /api/products/:product_id/related
+    axios.get(`/api/products/${productId}/related`)
+      .then(({ data }) => data)
+      .then((ids) => ids.map((id) => axios.get(`/api/products/${id}`)))
+      .then((promises) => {
+        Promise.all(promises).then((values) => setProducts(values));
+      });
+  }, []);
+
+  return (
+    <div className="related-products">
+      <h3>RelatedProducts</h3>
+      <div className="related-products__row">
+        {products.map((product) => <Card key={product.data.id} relatedProduct={product.data} />)}
```

Comment:
You might check products if it exists before it map function, something it throws an error it can not map of undefined.